source/tutorial/kotlin.txt

```diff
@@ -143,129 +174,223 @@ Modify the Application
 
-            val item = Item(realmApp.currentUser!!.identity)
-            item.priority = PriorityLevel.valueOf(spinner.selectedItem.toString()).priority
-            item.summary = itemSummaryInput.text.toString()
-            
-            CoroutineScope(Dispatchers.IO).launch {
-                realm.write {
-                    this.copyToRealm(item)
-                }
+            fun updateTaskPriority(taskPriority: PriorityLevel) {
+               _taskPriority.value = taskPriority
+            }
+
+            fun open() {
+               _expanded.value = true
             }
 
-      #. Finally, we will use the :android:`Toast <guide/topics/ui/notifiers/toasts>`
-         feedback to tell the user they have successfully created a new todo item with
-         their summary and priority. Within the ``Toast.makeText()`` method alter the
-         existing text to the following:
+            fun close() {
+               _expanded.value = false
+            }
+
+         Now update the ``addTask()`` and ``cleanUpAndClose()`` functions to include the new
+         ``taskPriority`` parameter and update the message to include the priority information:
+
+         .. code-block:: kotlin
+            :emphasize-lines: 4, 7
+
+            fun addTask() {
+               CoroutineScope(Dispatchers.IO).launch {
+                     runCatching {
+                        repository.addTask(taskSummary.value, taskPriority.value)
+                     }.onSuccess {
+                        withContext(Dispatchers.Main) {
+                           _addItemEvent.emit(AddItemEvent.Info("Task '$taskSummary' with priority '$taskPriority' added successfully."))
+                        }
+                     }.onFailure {
+                        withContext(Dispatchers.Main) {
+                           _addItemEvent.emit(AddItemEvent.Error("There was an error while adding the task '$taskSummary'", it))
+                        }
+                     }
+                     cleanUpAndClose()
+               }
+            }
 
          .. code-block:: kotlin
             :emphasize-lines: 3
 
-            Toast.makeText(
-                 this,
-                 "Item created: ${item.summary} with priority: ${item.priority}",
-                 Toast.LENGTH_SHORT
-             ).show()
+            private fun cleanUpAndClose() {
+               _taskSummary.value = ""
+               _taskPriority.value = PriorityLevel.Low
+               _addItemPopupVisible.value = false
+            }
+      
+      #. Finally, from the ``data`` folder, open the ``SyncRepository.kt`` file to 
+         reflect the same changes in the ``addTask()`` function, which writes the 
+         Item to the realm. 
+
+         First, add the following import below the ``package com.mongodb.app``:
+
+         .. code-block:: kotlin
+
+            import com.mongodb.app.domain.PriorityLevel
+
+         Next, update ``addTask()`` to pass the ``taskPriority`` as a parameter: 
```

Comment:
I think we should specify here _which_ version of `addTask()` we're telling people to update since it appears in the interface and also as a function we define below.
```suggestion
         Next, update ``addTask()`` in the `SyncRepository` interface to pass 
         the ``taskPriority`` as a parameter: 
```